src/libraries/Microsoft.Extensions.Caching.Abstractions/src/Hybrid/HybridCache.cs

```diff
@@ -0,0 +1,128 @@
+    /// Asynchronously sets or overwrites the value associated with the key.
+    /// </summary>
+    /// <typeparam name="T">The type of the data being considered.</typeparam>
+    /// <param name="key">The key of the entry to create.</param>
+    /// <param name="value">The value to assign for this cache entry.</param>
+    /// <param name="options">Additional options for this cache entry.</param>
+    /// <param name="tags">The tags to associate with this cache entry.</param>
+    /// <param name="cancellationToken">The <see cref="CancellationToken"/> used to propagate notifications that the operation should be canceled.</param>
+    public abstract ValueTask SetAsync<T>(string key, T value, HybridCacheEntryOptions? options = null, IReadOnlyCollection<string>? tags = null, CancellationToken cancellationToken = default);
+
+    /// <summary>
+    /// Asynchronously removes the value associated with the key if it exists.
+    /// </summary>
+    [System.Diagnostics.CodeAnalysis.SuppressMessage("ApiDesign", "RS0026:Do not add multiple public overloads with optional parameters", Justification = "Not ambiguous in context")]
+    public abstract ValueTask RemoveAsync(string key, CancellationToken cancellationToken = default);
+
+    /// <summary>
+    /// Asynchronously removes the value associated with the key if it exists.
+    /// </summary>
+    /// <remarks>Implementors should treat <c>null</c> as empty</remarks>
+    [System.Diagnostics.CodeAnalysis.SuppressMessage("ApiDesign", "RS0026:Do not add multiple public overloads with optional parameters", Justification = "Not ambiguous in context")]
+    public virtual ValueTask RemoveAsync(IEnumerable<string> keys, CancellationToken cancellationToken = default)
+    {
+        return keys switch
+        {
+            // for consistency with GetOrCreate/Set: interpret null as "none"
+            null or ICollection<string> { Count: 0 } => default,
+            ICollection<string> { Count: 1 } => RemoveAsync(keys.Single(), cancellationToken),
+            _ => ForEachAsync(this, keys, cancellationToken),
+        };
+
+        // default implementation is to call RemoveKeyAsync for each key in turn
+        static async ValueTask ForEachAsync(HybridCache @this, IEnumerable<string> keys, CancellationToken cancellationToken)
+        {
+            foreach (var key in keys)
+            {
+                await @this.RemoveAsync(key, cancellationToken).ConfigureAwait(false);
+            }
+        }
+    }
+
+    /// <summary>
+    /// Asynchronously removes all values associated with the specified tags.
+    /// </summary>
+    /// <remarks>Implementors should treat <c>null</c> as empty</remarks>
+    [System.Diagnostics.CodeAnalysis.SuppressMessage("ApiDesign", "RS0026:Do not add multiple public overloads with optional parameters", Justification = "Not ambiguous in context")]
+    public virtual ValueTask RemoveByTagAsync(IEnumerable<string> tags, CancellationToken cancellationToken = default)
```

Comment:
> Implementors should treat <c>null</c> as empty

```suggestion
    public virtual ValueTask RemoveByTagAsync(IEnumerable<string>? tags, CancellationToken cancellationToken = default)
```

Reply:
I don't think it makes sense to invite `null` inputs here